app/controllers/core_induction_programme/modules_controller.rb

```diff
@@ -1,7 +1,35 @@
 # frozen_string_literal: true
 
 class CoreInductionProgramme::ModulesController < ApplicationController
-  def show
-    @course_module = CourseModule.includes(:course_lessons).find(params[:module_id])
+  include Pundit
+  include GovspeakHelper
+
+  after_action :verify_authorized, except: :show
+  before_action :authenticate_user!, except: :show
+  before_action :load_and_authorize_course_module
+
+  def show; end
+
+  def edit
+    @module_preview = params[:module_preview] || @course_module.content
+    @html_content = content_to_html(@module_preview)
+  end
+
+  def update
+    if params[:commit] == "Save changes"
+      @course_module.update!(content: params[:module_preview])
```

Comment:
Just curious what should happen if updating fails for some reason.

Reply:
I imagine "nobody knows" is the answer. All of content editing is a bit speculative, and doesn't have designs - mainly because we will likely use it internally before release a lot more than it will be used afterwards.

So I think for time being "user sees an error caused by failed update" is fine.